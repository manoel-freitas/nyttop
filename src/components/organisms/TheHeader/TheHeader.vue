<template>
  <header :class="headerClass" @keyup.esc="openSideMenu = false">
    <the-top-header
      @onOpenSideMenu="openSideMenu = !openSideMenu"
    ></the-top-header>
    <the-side-menu
      v-show="openSideMenu"
      :show="openSideMenu"
      @onClose="openSideMenu = false"
    ></the-side-menu>
    <the-nav-bar v-if="($mq === 'laptop') | ($mq === 'desktop')" />
  </header>
</template>

<script>
import TheTopHeader from '../TheTopHeader'
import TheSideMenu from '../TheSideMenu'
import TheNavBar from '../TheNavBar'
export default {
  components: {
    TheTopHeader,
    TheSideMenu,
    TheNavBar
  },
  props: {
    headerClass: {
      type: String,
      required: false,
      default: ''
    }
  },
  data() {
    return {
      openSideMenu: false
    }
  }
}
</script>
