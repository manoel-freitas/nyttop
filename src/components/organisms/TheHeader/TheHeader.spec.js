/// <reference types="jest" />

import { shallowMount } from '@vue/test-utils'
import TheHeader from './TheHeader'

const theHeader = shallowMount(TheHeader, {
  stubs: ['ny-level', 'ny-level-item', 'ny-icon']
})

test('TheHeader has to be mounted', () => {
  expect(theHeader.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(theHeader.element).toMatchSnapshot()
})

test('should openSideMenu be false', () => {
  const theHeader = shallowMount(TheHeader, {
    stubs: ['ny-level', 'ny-level-item', 'ny-icon']
  })
  theHeader.setData({ openSideMenu: false })
  expect(theHeader.vm.openSideMenu).toBeFalsy()
})
