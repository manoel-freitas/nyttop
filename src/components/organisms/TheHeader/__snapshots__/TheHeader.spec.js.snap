// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<header
  class=""
>
  <the-top-header-stub />
   
  <the-side-menu-stub
    style="display: none;"
  />
   
  <!---->
</header>
`;
